ARCHITECTURE.md

```diff
@@ -0,0 +1,95 @@
+
+# Architecture
+
+This document should be updated semi-regularly. Feel free to open an issue if it hasn't been updated in more than a year.
+
+## Goals
+
+The major goal of Vello is to provide a high quality GPU accelerated renderer suitable for a range of 2D graphics applications, including rendering for GUI applications, creative tools, and scientific visualization.
+
+Vello emerges from being a research project, which attempts to answer these hypotheses:
+
+- To what extent is a compute-centered approach better than rasterization ([Direct2D])?
+- To what extent do "advanced" GPU features (subgroups, descriptor arrays, device-scoped barriers) help?
+- Can we improve quality and extend the imaging model in useful ways?
+
+Another goal of the overall project is to explain how the renderer is built, and to advance the state of building applications on GPU compute shaders more generally.
+Much of the progress on Vello is documented in blog entries.
+See [doc/blogs.md](doc/blogs.md) for pointers to those.
+
+
+## Roadmap
+
+The [roadmap for 2023](doc/roadmap_2023.md) is still largely applicable.
+The "Semi-stable encoding format" section and most of the "CPU fallback" section can be considered implemented.
+
+Our current priority is to fill in missing features and to fix rendering artifacts, so that Vello can reach feature parity with other 2D graphics engines.
+
+
+## File structure
+
+The repository is structured as such:
+
+- `crates/`
+  - `encoding/` - Types that represent the data that needs to be rendered.
+  - `shaders/` - Infrastructure to compile pipelines and shaders; see "Shader templating". Note that the `vello` crate doesn't currently import this crate (see #467).
```

Comment:
The #467 reference here and again a bit below should be links to the issue. They don't get auto-linked.

Reply:
I'd rather not. I'm in the mindset of keeping ARCHITECTURE.md easy to read in a text editor (unlike the README) so I don't want either inline links or deferred links.

I can say "issue #467" to be a bit more explicit.

Reply:
I agree with your point on this, Olivier; and I like the compromise of adding the word ‘issue’.

Reply:
What about:

```md
--- (see #467)
+++ [see #467]

or even just

+++ [#467]
```

I don't think that changing parentheses to brackets has a massive impact on the ease of reading in a text editor. However it allows us to define it as a link at the bottom of the file.